### PR TITLE
Fixes cspell not running for everyone consistently

### LINKED ISSUE #562: cspell ci
For some reason cspell doesn't seem to run for any one for the pre-commit hook. 

This bounty is for the cspell to be handled along the husky pre-commit hook.

### PR DESCRIPTION
**Pull request by BeanieMen**
Resolves #562
This pr is made to resolve cspell not running for everyone consistently by adding it to the build and test ci

### CODE REVIEW

**Comment by BeanieMen** [comment]
@pavlovcik can you review this?

**Comment by rndquu** [comment]
I don't really understand what issue this PR solves

So we had the [Spell Check workflow](https://github.com/ubiquity/ubiquity-dollar/blob/ed176304f4f89b57ec8975e7bf051261dc6a439f/.github/workflows/spell-check.yml) and now it is run inside the [Build & Test workflow](https://github.com/ubiquity/ubiquity-dollar/pull/717/files#diff-bc668a2c9f2299cef15b222055b4b4d5311646caec2e7610e540cee18ae9b948R72)

So technically nothing has changed

There is also an error on the "Spell Check" [step](https://github.com/ubiquity/ubiquity-dollar/actions/runs/5443658761/jobs/9900465509):
```
199/199 ./utils/kebabalize-typescript-imports.ts 7.46ms
CSpell: Files checked: 199, Issues found: 0 in 0 files
yes: standard output: Broken pipe
```

Besides in the original [issue description](https://github.com/ubiquity/ubiquity-dollar/issues/562#issue-1579772724) @pavlovcik says that the cspell check should be added to the build and test CI while later in the [comments](https://github.com/ubiquity/ubiquity-dollar/issues/562#issuecomment-1502131928) they say that `this bounty is for it to be handled within the husky pre-commit hook`. So again it is not clear how the issue should be resolved.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -3,13 +3,15 @@ name: Build & Test
 on:
   push:
   pull_request:
+    types: [opened, synchronize]
 
 jobs:
   build-and-test:
     name: Build & Test
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```
Comment (0x4007) [inline .github/workflows/build-and-test.yml:11]: Revert this. We don't want a dynamic environment or else our CI can randomly crash and it'll add complexity to debug.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -60,11 +63,17 @@ jobs:
...
       - uses: actions/upload-artifact@v3
         with:
           name: pr
           path: pr/
 
+      - name: Spell Check
+        run: |
+          LIST="$(find . -type f \( -name '*.sol' -o -name '*.tsx' -o -name '*.ts' \) ! -path '*/node_modules/*' ! -path '*/lib/*' ! -path '*/types/*')"
+          yes | npx cspell $LIST
+
```
Comment (0x4007) [inline .github/workflows/build-and-test.yml:76]: `spell-check.yml` should run all the time anyways so I don't see the purpose of you consolidating it inside of `build-and-test.yml`

**Review by 0x4007 — CHANGES_REQUESTED** [review]

**Comment by 0x4007** [comment]
> I don't really understand what issue this PR solves

Agreed I think this PR is unnecessary.

**Comment by rndquu** [comment]
> > I don't really understand what issue this PR solves
> 
> Agreed I think this PR is unnecessary.

Should we close the [issue](https://github.com/ubiquity/ubiquity-dollar/issues/562) as not planned or update the description to smth like "make sure husky pre-commit hook runs all the time" ?

**Comment by 0x4007** [comment]
Yes it looks like that was my mistake for not updating the spec from the comment below. Looks like the original specification had contradictory information.

**Pull request closed without merging**